### ISSUE TITLE
Zoom / Pan Listener or Camera Idle Callback

### ISSUE THREAD

**Issue by GeoffGodwin** [vasturiano/react-force-graph#39]
Is it possible to add a callback to the Graph so that it fires a callback on idle of the camera? When a user interacts via mouse scrolling and panning I'd like to actively memorize where they left off for later usage. Currently I have a zoom in and zoom out button in my own implementation and these keep track of zoom value but mouse panning/zooming disrupts this binding of zoom value.

Many thanks again for this excellent library!

**Comment by vasturiano** [comment]
@GeoffGodwin how do you define camera idle? Can you detect from the [trackball controls](https://github.com/JonLim/three-trackballcontrols), perhaps with the `end` event?
If so, you can access it using the `tbControls()` method. Try it, it may work. :)

**Comment by GeoffGodwin** [comment]
I'm actually using the 2D and I believe the tbControls() is only available in 3D unless I'm wrong?

**Comment by vasturiano** [comment]
@GeoffGodwin yes you're right, `tbControls` doesn't exist in 2D. It's a different zoom/panning mechanism, based on `d3-zoom`.
So, there isn't currently a way to detect pan/zoom user interactions from outside the component.
I can imagine adding an event notifier (say `onZoomPan()`) at every interaction, if there's enough interest in the feature.
For future reference, that should be triggered here:
https://github.com/vasturiano/force-graph/blob/master/src/force-graph.js#L327

**Comment by qasimalbaqali** [comment]
@vasturiano `onZoomPan()` would be an amazing feature if it gets added, I currently have a use case of limiting the max and min zoom of the canvas, will try with the above suggestion for now.
